pkg/addon-operator/ensure_crds.go

```diff
@@ -105,6 +105,46 @@ func (cp *CRDsInstaller) Run(ctx context.Context) *multierror.Error {
 	return result
 }
 
+func (cp *CRDsInstaller) DeleteCRDs(ctx context.Context, crdsToDelete []string) ([]string, error) {
+	var deletedCRDs []string
+	// delete crds listed in crdsToDelete if there are no related custom resources in the cluster
+	for _, crdName := range crdsToDelete {
+		deleteCRD := true
+		crd, err := cp.getCRDFromCluster(ctx, crdName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return nil, fmt.Errorf("error occurred during %s CRD clean up: %w", crdName, err)
+			}
+			continue
+		}
+
+		for _, version := range crd.Spec.Versions {
+			gvr := schema.GroupVersionResource{
+				Group:    crd.Spec.Group,
+				Version:  version.Name,
+				Resource: crd.Spec.Names.Plural,
+			}
+			list, err := cp.k8sClient.Resource(gvr).List(ctx, apimachineryv1.ListOptions{})
```

Comment:
It's useless to check resources in every version. API-server will convert all existent resource to the desired version. The best way is to fetch version with `storage: true` and check resources in that version